feat(core): add migration to remove entryComponents

crisbeto opened the pull request:
Adds an automated migration that will drop any usages of `entryComponents` from `@NgModule` and `@Component`.

crisbeto commented on packages/core/schematics/migrations.json:
```diff
@@ -9,6 +9,11 @@
       "factory": "./migrations/testbed-teardown/index"
+    },
+    "migration-v13.1-entry-components": {
+      "version": "13.1.0-beta",
```
We could. I suppose that I'll just have to bump the version here when we're close to 14?

devversion replied:
We could already have a second entry for v14 and keep the `13.1.0`-one as well. Not feeling strongly though.

crisbeto replied:
👍 I've added the second entry.

crisbeto committed:
feat(core): add migration to remove entryComponents

Adds an automated migration that will drop any usages of `entryComponents` from `@NgModule` and `@Component`.

dylhunn commented:
Merging by specific request -- ignoring Saucelabs checks

dylhunn commented:
This PR was merged into the repository by commit 5dff077d505da60ddead9f2ff2ddaaaea6feeb4a.
